hooks/001-extra-packages.chroot

```diff
@@ -151,3 +189,9 @@ esac
 apt install --no-install-recommends -y "${PACKAGES[@]}"
 
 apt autoremove -y
+
+# Copy snapd-generator, which will handle mounts for /lib/{firmware,modules}
+apt download snapd
+dpkg --fsys-tarfile snapd_*.deb |
+       tar xf - ./usr/lib/systemd/system-generators/snapd-generator
```

Comment:
Also `/usr/lib/systemd/system-environment-generators/snapd-env-generator` we should add, so we can remove the hacks from `hooks/022-setup-path.chroot`.
This could be done in another PR.